### PR TITLE
cart count was not updated after changing the amount of items in basket view

### PR DESCRIPTION
**Pull request by BoBaH83**
* fix cart->count recalculation after update items amount via cart
* fix misused comparison operator
* use array_key_exists instead of key_exists

### CODE REVIEW

File: Classes/Controller/CartController.php [hunk 1]
```diff
@@ -629,6 +629,7 @@ protected function sendBuyerMail(
         \Extcode\Cart\Domain\Model\Order\Address $billingAddress,
         \Extcode\Cart\Domain\Model\Order\Address $shippingAddress = null
     ) {
+	    /* @var \Extcode\Cart\Service\MailHandler $mailHandler*/
```
Comment (extcode) [inline Classes/Controller/CartController.php]: Please indent with spaces:
https://docs.typo3.org/typo3cms/CodingGuidelinesReference/PhpFileFormatting/GeneralRequirementsForPhpFiles/Index.html#whitespace-and-indentation

Please indent comments with the same number of spaces:
https://docs.typo3.org/typo3cms/CodingGuidelinesReference/PhpFileFormatting/PhpSyntaxFormatting/Index.html#comments

File: Classes/Controller/CartController.php [hunk 2]
```diff
@@ -522,10 +522,10 @@ public function initializeOrderCartAction()
         ) {
             $fields = $this->pluginSettings['validation']['orderCartAction']['fields'];
 
-            if (key_exists('acceptTerms', $fields)) {
+            if (array_key_exists('acceptTerms', $fields)) {
                 $this->setDynamicValidation('acceptTerms', $fields['acceptTerms']);
             }
-            if (key_exists('acceptConditions', $fields)) {
+            if (array_key_exists('acceptConditions', $fields)) {
```
Comment (extcode) [inline Classes/Controller/CartController.php]: key_exists is an alias for array_key_exists. but i will merge this.

File: Classes/Domain/Model/Cart/Cart.php [hunk 3]
```diff
@@ -1280,7 +1280,8 @@ private function calcTax()
      */
     public function reCalc()
     {
-        $this->calcGross();
+	    $this->calcCount();
```
Comment (extcode) [inline Classes/Domain/Model/Cart/Cart.php]: Do you habe an example for the wrong number of products in cart? I can imagine, when that happens, but would like to check the behavior.

Comment (extcode) [inline Classes/Domain/Model/Cart/Cart.php]: I used xdebug while changing the number of products in cart and can comprehend that. Please fix the indentation to four spaces instead one tab.

**Comment by extcode** [comment]
Thank you for your pull request. I add some notices. Can you add a new pull request with correct indentation for code and comments?

File: Classes/Controller/CartController.php (hunk 2 above)
Comment (BoBaH83) [inline Classes/Controller/CartController.php]: yes, but is shown as deprecated.
but the main reason is: you are already using array_key_exists on other places in code

File: Classes/Domain/Model/Cart/Cart.php (hunk 3 above)
Comment (BoBaH83) [inline Classes/Domain/Model/Cart/Cart.php]: I need shipping costs to be calculated per Item. After updating the amount in cart, i noticed that all the calculation was wrong. (taxes, shipping costs ...)

**Pull request merged**